pkg/kv/kvserver/replica_raft.go

```diff
@@ -1036,6 +1011,35 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 	} else if err != nil {
 		return stats, errors.Wrap(err, "checking raft group for Ready")
 	}
+
+	var outboundMsgs []raftpb.Message
+	var msgStorageAppend raftpb.Message
+	if hasReady {
+		logRaftReady(ctx, ready)
+		outboundMsgs, msgStorageAppend = splitLocalStorageMsgs(ready.Messages)
+	}
+	// Load the committed entries to be applied after releasing Replica.mu, to
+	// ensure that we don't have IO under this narrow/lightweight mutex. The
+	// RawNode can be making progress in the meantime, but it will never overwrite
+	// the committed entries it has been observing during the Ready() call.
+	var toApply []raftpb.Entry
+	if !ready.Committed.Empty() {
```

Comment:
Should move this all down to right before `var appTask apply.Task` creation. The benefit would be that `r.sendRaftMessages(ctx, outboundMsgs)` are sent eagerly, which reduces latency by that of reading from log storage.

Reply:
Done.